tasks/abs.rb

```diff
@@ -25,11 +25,17 @@ def provision(platform, inventory_location, vars)
 
   headers = { 'X-AUTH-TOKEN' => token_from_fogfile('abs'), 'Content-Type' => 'application/json' }
   priority = (ENV['CI']) ? 1 : 2
-  payload = { 'resources' => { platform => 1 },
-              'priority' => priority,
-              'job' => { 'id' => job_id,
-                         'tags' => { 'user' => Etc.getlogin, 'jenkins_build_url' => jenkins_build_url } } }
-
+  if !platform.nil?
+    payload = { 'resources' => { platform => 1 },
+		'priority' => priority,
+		'job' => { 'id' => job_id,
+			   'tags' => { 'user' => Etc.getlogin, 'jenkins_build_url' => jenkins_build_url } } }
```

Comment:
I'm wondering could we ditch the old way altogether?
```
platforms = [platform] if platform
```
Then just use the `payload` in the `else` block. Or is there benefit in retaining the old way too that I'm missing?